__tests__/List.ts

```diff
@@ -484,6 +484,24 @@ describe('List', () => {
     expect(r.toArray()).toEqual(['b', 'd', 'f']);
   });
 
+  it('filters values based on type', () => {
+    class A {}
+    class B extends A {
+      b(): void {}
+    }
+    class C extends A {
+      c(): void {}
+    }
+    let l1 = List<A>([ new B(), new C(), new B(), new C() ]);
+    // Note: TypeScript requires a defined type guard to reduce the type. Since
+    // arrow functions cannot define return types, we must use a normal function.
+    let l2: List<C> = l1.filter(function (v): v is C {
```

Comment:
Oh, I didn't that you can workaround that `as` by just using a `function`, nice!